scripts/irods/test/test_iadmin.py

```diff
@@ -2535,3 +2535,159 @@ def test_mkgroup_with_remote_zone(self):
+    def test_bad_nohost(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'badzone', 'remote', ':1234'],
+                                       'STDERR', 'CAT_HOSTNAME_INVALID')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmzone', 'badzone'])
+
+    def test_bad_negport(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'badzone', 'remote', 'hostname:-1234'],
+                                       'STDERR', 'CAT_INVALID_ARGUMENT')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmzone', 'badzone'])
+
+    def test_bad_bigport(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'badzone', 'remote', 'hostname:1234567890'],
+                                       'STDERR', 'CAT_INVALID_ARGUMENT')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmzone', 'badzone'])
+
+    def test_bad_NaNportNaN(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'badzone', 'remote', 'hostname:asdf1234asdf'],
+                                       'STDERR', 'CAT_INVALID_ARGUMENT')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmzone', 'badzone'])
+
+    def test_bad_NaNport(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'badzone', 'remote', 'hostname:asdf1234'],
+                                       'STDERR', 'CAT_INVALID_ARGUMENT')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmzone', 'badzone'])
+
+    def test_bad_portNaN(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'badzone', 'remote', 'hostname:1234asdf'],
+                                       'STDERR', 'CAT_INVALID_ARGUMENT')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmzone', 'badzone'])
+
+class test_modzone_conn_str_validation(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up the test class."""
+        cls.admin = session.mkuser_and_return_session('rodsadmin', 'otherrods', 'rods', lib.get_hostname())
+
+    @classmethod
+    def tearDownClass(cls):
+        """Tear down the test class."""
+        with session.make_session_for_existing_admin() as admin_session:
+            cls.admin.__exit__()
+            admin_session.assert_icommand(['iadmin', 'rmuser', cls.admin.username])
+
+    def setUp(self):
+        self.admin.assert_icommand(['iadmin', 'mkzone', 'goodzone_localhost', 'remote', 'localhost:1247'], 'EMPTY')
+        self.admin.assert_icommand(['iadmin', 'mkzone', 'goodzone_blank', 'remote', 'localhost:1247'], 'EMPTY')
+
+    def tearDown(self):
+        self.admin.run_icommand(['iadmin', 'rmzone', 'goodzone_localhost'])
+        self.admin.run_icommand(['iadmin', 'rmzone', 'goodzone_blank'])
+
+    def test_good_localhost(self):
+        self.admin.assert_icommand(['iadmin', 'modzone', 'goodzone_blank', 'conn', 'localhost:1247'], 'EMPTY')
```

Comment:
Should we confirm that the zone was modified successfully (or not) based on the success (or failure) of `modzone`?

Reply:
Good idea